### PR TITLE
Editorial pass and author update

### PR DESCRIPTION
**Pull request by LPardue**
My first intent was to fix typos but then I got carried away. I can split into two PRs if the changes are too contentious.

### CODE REVIEW

File: draft-kazuho-httpbis-priority.md [hunk 1]
```diff
@@ -116,22 +123,23 @@ Argument syntax:
...
 * `non-blocking` indicates that the response does not block the processing of
-  the document even though the response is being used or referred by the
+  the document even though the response is being used or referred to by the
   document.
 
-When the Priority header field or the `urgency` directive do not appear in the
-request, the server SHOULD act as if urgency level of `document` was specified.
+The default value of the `urgency` directive is `document`. This SHOULD apply if
+the directive is omitted, if the Priority header is missing, or if the
+server ignores the Priority header field.
```
Comment (kazuho) [inline draft-kazuho-httpbis-priority.md]: Could you please clarify the intent of stating "or if the server ignores the Priority header field"?

I am not sure what it suggests, and is confusing.

File: draft-kazuho-httpbis-priority.md [hunk 2]
```diff
@@ -153,31 +161,31 @@ progressive = "yes" / "no"
 This boolean directive indicates if a response can be processed progressively,
 i.e. provide some meaningful output as chunks of the response arrive.
 
-When the Priority header field or the `progressive` directive do not appear in
-the request, the server SHOULD act as if a `progress` directive with an argument
-of `no` has specified.
+The default value of the `progressive` directive is `no`. This SHOULD apply if
+the directive is omitted, if the Priority header is missing, or if the
+server ignores the Priority header field.
```
Comment (kazuho) [inline draft-kazuho-httpbis-priority.md]: Here too.

File: draft-kazuho-httpbis-priority.md [hunk 3]
```diff
@@ -94,12 +100,13 @@ interpreted as described in [RFC2119].
 
 # The Priority HTTP Header Field
 
-The Priority HTTP header field is used by a client as part of an HTTP request
-to specify the priority of the response.  A server can also use the header field
-as part of an HTTP request to communicate the correct or amended precedence to
-an intermediary.
+The Priority HTTP header field may appear in requests and responses. A client
```
Comment (kazuho) [inline draft-kazuho-httpbis-priority.md]: How about s/may/can/? "May" is a RFC2119 word.

**Review by kazuho — COMMENTED** [review]
This is much much better. Thank you very much for the PR.

Looks good to me aside from the three issues below.

File: draft-kazuho-httpbis-priority.md (hunk 1 above)
Comment (kazuho) [reply draft-kazuho-httpbis-priority.md]: PS. Now I realize that this might be due to the provision we have in the "Coexistence with HTTP/2" section?

Assuming that is true, and through the discussion in #2, I think we might want to downgrade the "Coexistence ~" section to one of considerations, and remove this "if ignores..." provision for simplicity. Using a boolean settings parameter to disable header-based prioritization is just one way of achieving coexistence, and I think there is room to explore other designs.

WDYT?

Comment (LPardue) [reply draft-kazuho-httpbis-priority.md]: While editing from start to finish i built up more understanding of the spec. However, I think I'm still a little confused about exactly when the Priority header can appear on requests r responses. The reason being that we state the header is "HTTP version independent". Can this header only appear on requests/responses that are made over HTTP/2 and HTTP/3? What happens if the header is present but the setting is not advertised? Is the priority ignored in favor of frame-based prioritization? Conversely, we state that a default priority applies if the header is missing, but that is only meaningful if the server was expecting to see the header; this leads me to believe that it relies upon negotiation via Settings.

On my first reading it seemed feasible that a HTTP/1.x client could add the Priority header to the requests that it makes. If it makes multiple requests over separate connections, and the server is able to link those connections somehow, the priority information could be used to manage the precedence of responses across all connections. This thinking also somewhat applies to a protocol translation proxy that talks h2 to the client but HTTP/1.1 to the server. The header is end-to-end (answered on #4) so what does the server do with that header in this case? I think the answer then falls to HTTP core semantics of ignoring unknown headers, in which case we should probably not say anything about a default priority in ignoring case, like you suggest.

File: draft-kazuho-httpbis-priority.md (hunk 3 above)
Comment (LPardue) [reply draft-kazuho-httpbis-priority.md]: oops, fixed thanks

File: draft-kazuho-httpbis-priority.md (hunk 1 above)
Comment (kazuho) [reply draft-kazuho-httpbis-priority.md]: Regarding what we need to specify in this document, it is my understanding that we only need to talk about endpoints that implement this "extension". We will not be changing the behavior of endpoints that do not implement this, and we should not (unless it is a clarification), because we can't, as we are not updating RFC 723x or 7540.

> I think the answer then falls to HTTP core semantics of ignoring unknown headers, in which case we should probably not say anything about a default priority in ignoring case, like you suggest.

:+1: I think this is the right thing to do.

If we need to make changes to the "Coexistence with HTTP/2 ~" for better clarification can be a separate discussion, but now I realize that it might be better to clarify the rationale a bit more. Below is what I have in mind. Does it make sense to you?

Because the header is end-to-end, it should be used without any negotiation. However, we need to have an exception for HTTP/2, because there is already a different prioritization mechanism _that has wider range of scope_, and we need to provide guidance for transition (or coexistence).

In short term, H2 clients would use both mechanisms, and that would work mostly fine. A server that recognizes both can use either (though we recommend using the header-based scheme), with one exception. The exception is when the client is in fact a proxy that coalesces connections from clients. This is the exact case that we do not cover, and therefore we define a way of ignoring Priority header field in favor of the H2 way.

Comment (LPardue) [reply draft-kazuho-httpbis-priority.md]: SGTM, I'll update the PR to remove reference to ignoring

**Comment by LPardue** [comment]
Removed use of ignoring header, squashed and updated the PR.

**Review by kazuho — APPROVED** [review]

**Comment by kazuho** [comment]
Thank you!
